packages/main/src/RPA/MSGraph.py

```diff
@@ -0,0 +1,956 @@
+
+            Download folder with object
+                ${download_path}=    Download Folder From Onedrive
+                ...    ${drive_item}
+                ...    /path/to/local/folder
+        """
+        self._require_authentication()
+        drive_instance = self._get_drive_instance(resource, drive_id)
+        folder_instance = self._get_folder_instance(drive_instance, target_folder)
+        return self._download_folder(folder_instance, to_path)
+
+    @keyword
+    def find_onedrive_file(
+        self,
+        search_string: str,
+        target_folder: Union[drive.Folder, str, None] = None,
+        include_folders: Optional[bool] = False,
+        resource: Optional[str] = None,
+        drive_id: Optional[str] = None,
+    ) -> list[drive.DriveItem]:
+        # pylint: disable=anomalous-backslash-in-string
+        """Returns a list of files found in OneDrive based on the search string.
+        If a folder is not specified, the search is done in the entire drive and
+        may include items that were shared with the user. It is possible to pass
+        \`root\` as the target folder in order to avoid this behavior.
+
+        The files returned are DriveItem objects and they have additional
+        properties that can be accessed with dot-notation, see
+        \`List Files In Onedrive Folder\` for details.
+
+        :param search_string: String used to search for file in OneDrive.
+         Values may be matched across several fields including filename,
+         metadata, and file content.
+        :param target_folder: Folder where to search for files.
+        :param include_folders: Boolean indicating if should return folders as well.
+        :param resource: Name of the resource if not using default.
+        :param drive_id: Drive ID if not using default.
+        :return: List of DriveItems found based on the search string.
+
+        .. code-block: robotframework
+
+            *** Tasks ***
+            Find file
+                ${files}=    Find Onedrive File    Report.xlsx
+        """  # noqa: W605
+        self._require_authentication()
+        drive_instance = self._get_drive_instance(resource, drive_id)
+        if target_folder:
+            folder = self._get_folder_instance(drive_instance, target_folder)
+            items = folder.search(search_string)
+        else:
+            items = drive_instance.search(search_string)
+        if include_folders:
+            return items
+        return [item for item in items if not item.is_folder]
+
+    @keyword
+    def download_file_from_share_link(
+        self,
+        share_url: str,
+        to_path: Union[Path, str, None] = None,
+        name: Optional[str] = None,
+    ) -> Path:
+        """Downloads file from the share link.
+
+        The downloaded file will be saved to a local path.
+
+        :param share_url: URL of the shared file
+        :param to_path: Destination folder of the downloaded file,
```

Comment:
possible formatting problem (I think one space is enough for the continuation line)